tensorboard/plugins/graph/tf_graph_common/util.ts

```diff
@@ -106,14 +113,25 @@ module tf.graph.util {
     // UI to update.
     try {
       let result = tf.graph.util.time(msg, task);
-      // Update the progress value.
-      tracker.updateProgress(incProgressValue);
+
+      if (result instanceof Promise) {
+        result.then((val) => {
+          tracker.updateProgress(incProgressValue);
+          return val;
+        }, (e) => {
+          tracker.reportError('Failed ' + msg, e);
+        });
+      } else {
+        // Update the progress value.
+        tracker.updateProgress(incProgressValue);
+      }
       // Return the result to be used by other tasks.
       return result;
     } catch (e) {
       // Errors that happen inside asynchronous tasks are
       // reported to the tracker using a user-friendly message.
       tracker.reportError('Failed ' + msg, e);
+      throw e;
```

Comment:
Why throw? The whole point of these utilities is to have nicely
formatted errors rather than dumping stack traces to the console, right?

Reply:
Was related to the "Promise shenanigans". Since we never swallow Promise rejection, I wanted to take it congruent. Removed after removing those Promise checks.